src/TextInput/TextInput.tsx

```diff
@@ -134,7 +134,7 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
           showLoadingIndicator={showLeadingLoadingIndicator}
           hasLoadingIndicator={typeof loading === 'boolean'}
         >
-          {typeof LeadingVisual === 'function' ? <LeadingVisual /> : LeadingVisual}
+          {LeadingVisual && <LeadingVisual></LeadingVisual>}
```

Comment:
Refactor needed due to removing support for `Octicon` ([reference](https://github.com/primer/octicons/pull/943/files))